pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -157,15 +160,20 @@ public void addSegment(ImmutableSegment segment) {
         "Got unsupported segment implementation: {} for segment: {}, table: {}", segment.getClass(), segmentName,
         _tableNameWithType);
     ImmutableSegmentImpl immutableSegment = (ImmutableSegmentImpl) segment;
+    double maxComparisonValue =
```

Comment:
(MAJOR) We cannot read value here because there is no guarantee the comparison column is number when TTL is not enabled

Reply:
Thanks for pointing this out! Updated!